src/porepy/numerics/interface_laws/elliptic_interface_laws.py

```diff
@@ -429,15 +404,16 @@ def _assemble(
 
                 rhs[2] = rhs[2] - vector_source_discr * vector_source
 
-            rhs[2] = matrix_dictionary_edge[self.mortar_scaling_matrix_key] * rhs[2]
+            rhs[2] = diffusivity_discr * rhs[2]
 
         if assemble_matrix:
-            for block in range(cc.shape[1]):
-                # Scale the pressure blocks in the mortar problem
-                cc[2, block] = (
-                    matrix_dictionary_edge[self.mortar_scaling_matrix_key]
-                    * cc[2, block]
-                )
+            for block in range(3):
+                # Scale the pressure contributions in the mortar equation by
+                # normal diffusivity.
+                cc[2, block] = diffusivity_discr * cc[2, block]
```

Comment:
diffusivity_discr is diagonal scaling with volume and normal diffusivity? If so, it should not hit `cc[2, 2]`?

Reply:
.. I see below that you have not yet added it. Still, I would prefer if we let block go in `range(2)`, and add a comment that the mortar block should not be scaled.

Reply:
If only. But block 2 has to be hit prior to addition of "own" mortar flux due to contributions from interface fluxes to trace of primary pressure.